python/pointmatcher/io.cpp

```diff
@@ -0,0 +1,217 @@
+#include "io.h"
+
+#include "pointmatcher/IO.h"
+
+PYBIND11_MAKE_OPAQUE(PointMatcherIO<ScalarType>::FileInfoVector)
+
+namespace pointmatcher
+{
+	void pybindIO(py::module& p_module)
+	{
+		using PMIO = PointMatcherIO<ScalarType>;
+		py::class_<PMIO> pyPointMatcherIO(p_module, "PointMatcherIO", "IO Functions and classes that are dependant on scalar type are defined in this templatized class");
+
+		pyPointMatcherIO
+			.def(py::init<>())
+			.def_static("getColLabel", &PMIO::getColLabel, py::arg("label"), py::arg("row"),"convert a descriptor label to an appropriate sub-label");
+
+		using PMPropTypes = PMIO::PMPropTypes;
+		py::enum_<PMPropTypes>(pyPointMatcherIO, "PMPropTypes", "Type of information in a DataPoints. Each type is stored in its own dense matrix.")
+			.value("FEATURE", PMPropTypes::FEATURE)
+			.value("DESCRIPTOR", PMPropTypes::DESCRIPTOR)
+			.value("TIME", PMPropTypes::TIME)
+			.value("UNSUPPORTED", PMPropTypes::UNSUPPORTED);
+
+		using SupportedLabel = PMIO::SupportedLabel;
+		py::class_<SupportedLabel>(pyPointMatcherIO, "SupportedLabel", "Structure containing all information required to map external information to PointMatcher internal representation")
+			.def_readwrite("internalName", &SupportedLabel::internalName, "name used in PointMatcher")
+			.def_readwrite("externalName", &SupportedLabel::externalName, "name used in external format")
+			.def_readwrite("type", &SupportedLabel::type, "type of information in PointMatcher")
+
+			.def(py::init<const std::string&, const std::string&, const PMPropTypes&>(), py::arg("internalName"), py::arg("externalName"), py::arg("type"));
+
+		using SupportedLabels = PMIO::SupportedLabels;
+		py::bind_vector<SupportedLabels>(pyPointMatcherIO, "SupportedLabels", "Vector of supported labels in PointMatcher and their external names");
+
+		using GenericInputHeader = PMIO::GenericInputHeader;
+		py::class_<GenericInputHeader>(pyPointMatcherIO, "GenericInputHeader", "Helper structure designed to parse file headers")
+			.def_readwrite("name", &GenericInputHeader::name)
+			.def_readwrite("matrixRowId", &GenericInputHeader::matrixRowId)
+			.def_readwrite("matrixType", &GenericInputHeader::matrixType)
+			.def(py::init<>())
+			.def(py::init<const std::string&>(), py::arg("name"));
+
```

Comment:
Could you add comments describing what each of the above blocks do?

Reply:
Yes, I will eventually add comments to describe the whole thing. I also thought maybe I could add a section to the tutorials with a link to the pybind11 documentation for those who would like to contribute to the bindings.

Reply:
Probably in this PR, but it can go in another one as well.